Sources/Fluent/Fluent+Response.swift

```diff
@@ -0,0 +1,106 @@
+    /// Creates a `Response` object that has a status of 201 (Created) and includes the `Location` HTTP header.
+    ///
+    /// Intended to be used in routes that create a new object.
+    ///
+    /// ### Note ###
+    /// The return type of your route should be `EventLoopFuture<Response>`.
+    ///
+    /// ### Example ###
+    /// ```swift
+    /// func create(req: Request) throws -> EventLoopFuture<Response> {
+    ///    let todo = try req.content.decode(Todo.self)
+    ///    return todo.create(on: req.db).flatMapThrowing {
+    ///        try .created(todo, for: req)
+    ///    }
+    /// }
+    /// ```
+    /// - Parameters:
+    ///   - obj: The newly created model.
+    ///   - request: The `Request` which is being responded to.
+    ///   - mediaType: How to encode the response, defaulting to `.json`
+    /// - Throws: If the object hasn't been created or the encoding fails.
+    /// - Returns: A `Response` object.
+    public static func created<T>(_ obj: T,
+                                  for request: Request,
+                                  as mediaType: HTTPMediaType = .json) throws -> Response
+        where T: Model {
+            let id = try obj.requireID()
+            let location = "\(request.url.string)/\(id)"
+
+            return try created(obj, location: location, as: mediaType)
+    }
+
+    /// Creates a `Response` object that has a status of 201 (Created) and includes the `Location` HTTP header.
+    ///
+    /// Intended to be used in routes that create a new object. Many models contain the actual `id` which
+    /// is the primary key in the database as an `Int`, but then also include a public identifier like a `UUID`
+    /// that consumers utilize. This is simply a convenience method to return that public ID in the `Location` header.
+    ///
+    /// ### Note ###
+    /// The return type of your route should be `EventLoopFuture<Response>`.
+    ///
+    /// ### Example ###
+    /// ```swift
+    /// func create(req: Request) throws -> EventLoopFuture<Response> {
+    ///    let todo = try req.content.decode(Todo.self)
+    ///    return todo.create(on: req.db).flatMapThrowing {
+    ///        try .created(todo, for: req, publicID: todo.publicUUID.uuidString)
+    ///    }
+    /// }
+    /// ```
+    /// - Parameters:
+    ///   - obj: The newly created model.
+    ///   - request: The `Request` which is being responded to.
+    ///   - publiciD: The user visible ID that the model should be queried against.
+    ///   - mediaType: How to encode the response, defaulting to `.json`
+    /// - Throws: If the object hasn't been created or the encoding fails.
+    /// - Returns: A `Response` object.
+    public static func created<T, ID>(_ obj: T,
+                                  for request: Request,
+                                  publicID: ID,
+                                  as mediaType: HTTPMediaType = .json) throws -> Response
+        where T: Model, ID: StringProtocol {
```

Comment:
Could we use `CustomStringConvertible` here to support things like `UUID`?

Reply:
This should be `T: Content` as well since we're not using anything `Model` specific here.